astropy/modeling/fitting.py

```diff
@@ -1141,7 +1149,8 @@ def _convert_input(x, y, z=None, n_models=1, model_set_axis=0):
             # That is, each model should be represented by a column.  TODO:
             # Obviously this is a detail of np.linalg.lstsq and should be
             # handled specifically by any fitters that use it...
-            y = np.rollaxis(y, model_set_axis, y.ndim)
+            if rollaxis:
+                y = np.rollaxis(y, model_set_axis, y.ndim)
```

Comment:
I think I need to put this in the LinearLSQFitter instead, as suggested by the comment.